dart-common/src/main/java/dart/common/util/NavigationModelBindingTargetUtil.java

```diff
@@ -82,28 +82,33 @@ private void checkForParentsOutside(
       final NavigationModelBindingTarget navigationModelBindingTarget = target.getValue();
       // root inside module
       if (navigationModelBindingTarget.parentPackage == null) {
-        TypeElement ancestorElement = element;
-        while (true) {
-          final TypeMirror superType = ancestorElement.getSuperclass();
-          if (superType.getKind() == TypeKind.NONE) {
-            // Got to the oldest ancestor and none contains NavigationModel field
-            return;
-          }
-          ancestorElement = (TypeElement) ((DeclaredType) superType).asElement();
-          // ancestor contains a NavigationModel field
-          if (getNavigationModelBinder(ancestorElement) != null) {
-            navigationModelBindingTarget.parentPackage =
-                compilerUtil.getPackageName(ancestorElement);
-            navigationModelBindingTarget.parentClass =
-                compilerUtil.getClassName(
-                    ancestorElement, navigationModelBindingTarget.parentPackage);
-            return;
-          }
+        TypeElement ancestorElement = findClosestAncestorWithNavigationModelBinder(element);
```

Comment:
`ancestorNavigationModelBinderTypeElement`

Reply:
after discussing offline, it is the ancestor and not it model